### PR TITLE
Sk ja empty pages

### LINKED ISSUE #1: Create empty pages and navbar component
- Make empty pages for what's new, contact us, and login page
- Add pages to frontend router
- Unstyled navbar component with links to each page

### PR DESCRIPTION
**Pull request by saahilkumar**
Closes #1 

- Added three new empty pages
  - Login -- `/login`
  - What's New -- `/whatsnew`
  - Contact Us -- `/contact`
- Created an unstyled NavBar component that links between these three pages 

**Question:** Should we add the NavBar to the main page (`index.tsx`)? Right now, it's only on the three new pages.

Here's a screenshot of the Contact Us page for example:

Screen Shot 2022-01-31 at 5 15 15 PM

### CODE REVIEW

File: lib/components/navbar/NavLink.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Link from "next/link";
+import { withRouter } from "next/router";
+
+const NavLink = props => (
+  <Link href={props.path}>
+    <div
+      className={`NavButton ${
+        props.router.pathname === props.path ? "active" : ""
```
Comment (rymaju) [inline lib/components/navbar/NavLink.tsx:8]: 📌 This is interesting, although we will eventually want to replace this with Chakra UI components

File: lib/components/navbar/links.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const navLinks = [
```
Comment (rymaju) [inline lib/components/navbar/links.tsx:1]: 👍 I like this!

File: pages/contact/index.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import type { NextPage } from 'next';
+import NavBar from "../../lib/components/navbar";
+import navLinks from "../../lib/components/navbar/links";
+
+const ContactUs: NextPage = () => {
+
+    return (
+      <>
+      <NavBar navLinks={navLinks} />
```
Comment (rymaju) [inline pages/contact/index.tsx:9]: ❓ This looks suspiciously not formatted correctly, seems like we want another level of indentation on this line. Did you run prettier?

**Review by rymaju — APPROVED** [review]
Looks good but I think there are some minor formatting issues, are you sure you ran prettier?

**Comment by rymaju** [comment]
image
This is what I got from `npm prepush:fix`, looks like there are a couple things to fix, as well as some existing things that got picked up from prettier.

Not sure how that got skipped on your end? Do you get this too? @saahilkumar ?

**Comment by saahilkumar** [comment]
> Not sure how that got skipped on your end? Do you get this too? @saahilkumar ?

No I didn't see that on my end D: might be a node version thing because I don't think I updated to 17.3.0 yet

**Comment by rymaju** [comment]
hmm weird, I can fix it for you for now

File: lib/components/navbar/links.tsx (hunk 2 above)
Comment (varunthakkar1) [reply lib/components/navbar/links.tsx:1]: This is super smart, love it!

**Review by varunthakkar1 — APPROVED** [review]
LGTM! Nice work, just address comments please 👍🏾

**Comment by saahilkumar** [comment]
@rymaju @varunthakkar1

I replaced `props` with `props: any` in a few different places to get rid of the build errors. Is that chill or is that bad style?

**Comment by rymaju** [comment]
any is considered harmful 😠

we 'never' talk to him anymore, its like hes someone totally different -- completely 'unknown'

Nobody else will think that was funny but I find my own typescript puns absolutely hilarious

**Comment by rymaju** [comment]
We want to declare the actual type of props and then use that instead of any, I can find an example somewhere... or you can take a look at the other frontend components in other projects yourself

**Comment by rymaju** [comment]
Heres an example: https://github.com/Code-4-Community/jpal-frontend/blob/master/src/components/survey/ConfirmAssignments.tsx

File: lib/components/navbar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import NavLink from './NavLink';
+
+const NavBar = (props: any) => (
```
Comment (rymaju) [inline lib/components/navbar/index.tsx:3]:
```suggestion
const NavBar: React.FC<NabBarProps> = ({navLinks}) => (
```

File: lib/components/navbar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  <div className="NavBar">
+    {props.navLinks.map((button: any) => (
```
Comment (rymaju) [inline lib/components/navbar/index.tsx:5]: 🔧 This should have a type! Is it something like:
```ts
type NavBarButton = {
    path: string,
    label: string
}
```
?

File: lib/components/navbar/links.tsx (hunk 2 above)
Comment (rymaju) [inline lib/components/navbar/links.tsx:1]: Ah, this is where you want to define and export your NavBar types

File: lib/components/navbar/NavLink.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+const NavLink = (props: any) => (
+  <Link href={props.path} passHref>
+    <div className={`NavButton ${props.router.pathname === props.path ? 'active' : ''}`}>
+      <span className="Label">{props.label}</span>
+    </div>
+  </Link>
+);
+
+export default withRouter(NavLink);
```
Comment (rymaju) [inline lib/components/navbar/NavLink.tsx:12]: 🤔 Why not use the `useRouter` hook? https://nextjs.org/docs/api-reference/next/router

File: lib/components/navbar/NavLink.tsx [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import Link from 'next/link';
+import { withRouter } from 'next/router';
+
+const NavLink = (props: any) => (
```
Comment (rymaju) [inline lib/components/navbar/NavLink.tsx:4]:
```suggestion
const NavLink : React.FC<NavLinkProps>= ({router, path, label}) => (
```
Or even better, reuse your already defined type:
```suggestion
const NavLink : React.FC<NavBarButton>= ({path, label}) => (
```
and then use `useRouter` in the body of the function.

Or, if you really want to use `withRouter` for some reason:
```suggestion
const NavLink : React.FC<NavBarButton & { NextRouter }>= ({router, path, label}) => (
```
Idk if its called `NextRouter` but you get the gist.

File: pages/contact/index.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import type { NextPage } from 'next';
+import NavBar from '../../lib/components/navbar';
+import navLinks from '../../lib/components/navbar/links';
+
+const ContactUs: NextPage = () => {
```
Comment (rymaju) [inline pages/contact/index.tsx:5]: ❓ What is a `NextPage`? Why should we use it over a `React.FC`?

**Review by rymaju — CHANGES_REQUESTED** [review]
😓 Sorry I looked through this and realized I have a lot more questions, maybe we can hop on a call to go over everything? @saahilkumar @ja0911

**Pull request merged**